pkg/miscellaneous/miscellaneous.go

```diff
@@ -338,6 +338,36 @@ func ProbeEtcd(ctx context.Context, clientFactory etcdClient.Factory, logger *lo
 	return nil
 }
 
+// IsMultiNode determines whether a pod is part of a multi node setup or not
+// This is determined by checking the `initial-cluster` of the etcd configmap to check the number of members expected.
+func IsMultiNode() bool {
+	var inputFileName string
+	etcdConfigForTest := os.Getenv("ETCD_CONF")
+	if etcdConfigForTest != "" {
+		return false
+	}
+	inputFileName = "/var/etcd/config/etcd.conf.yaml"
```

Comment:
This should be transferred into a utility function. I also see this code being duplicated multiple times in the code base.

Reply:
Done

Reply:
Can you please also make other areas in the code use this utility?

- https://github.com/gardener/etcd-backup-restore/blob/a6ee3fd56bc0401fe54f048fe8166750f677a777/pkg/miscellaneous/miscellaneous.go#L295
- https://github.com/gardener/etcd-backup-restore/blob/a6ee3fd56bc0401fe54f048fe8166750f677a777/pkg/member/member_control.go#L74
- https://github.com/gardener/etcd-backup-restore/blob/a6ee3fd56bc0401fe54f048fe8166750f677a777/pkg/server/backuprestoreserver.go#L103

Reply:
Done